cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerServiceTest.java

```diff
@@ -165,6 +168,41 @@ public void testSimpleSchedulerLifecycle() throws Exception {
     checkState(Scheduler.ScheduleState.NOT_FOUND, scheduleIds);
   }
 
+  @Test
+  public void testScheduleUpdate() throws Exception {
+    AppFabricTestHelper.deployApplication(namespace, AppWithWorkflow.class);
+    ApplicationSpecification applicationSpecification = store.getApplication(appId);
+
+    schedulerService.schedule(program, programType, ImmutableList.of(timeSchedule1));
+    applicationSpecification = createNewSpecification(applicationSpecification, program, programType, timeSchedule1);
+    store.addApplication(appId, applicationSpecification, locationFactory.create("app"));
```

Comment:
@anew this is because the scheduler APIs currently do not do any store updates. They expect the callers to handle store operations. We have https://issues.cask.co/browse/CDAP-1202 open for this. I had a PR - https://github.com/caskdata/cdap/pull/3084 to fix this, but we decided to not do it in 3.1, because it would need a bigger refactoring of the scheduler classes. 

Basically, the problem is that currently, ``AbstractSchedulerService``, and ``TimeScheduler`` and ``StreamSizeScheduler`` implement the same interface, but really have different responsibilities. So while doing this, we ended up expanding the scheduler apis with more operations that had to throw UnsupportedOperationException in some implementations.

We should pick up that PR again, but I haven't gotten to it since 3.1